packages/react-components/react-field/src/components/Field/useFieldStyles.ts

```diff
@@ -12,58 +12,30 @@ export const getFieldClassNames = (name: string): SlotClassNames<FieldSlots<Fiel
   hint: `fui-${name}__hint`,
 });
 
+// Size of the icon in the validation message
+const iconSize = '12px';
+
 /**
  * Styles for the root slot
  */
 const useRootStyles = makeStyles({
   base: {
     display: 'grid',
-    gridAutoFlow: 'row',
-    gridTemplateColumns: 'auto 1fr',
-    gridTemplateAreas: `
-      "label label"
-      "control control"
-      "validationIcon validationMessage"
-      "hint hint"
-    `,
     justifyItems: 'start',
   },
 
+  // In horizontal layout, the field is a grid with the label taking up the entire first column.
+  // The last row is slack space in case the label is taller than the rest of the content.
   horizontal: {
-    gridTemplateColumns: '33% auto 1fr',
-    gridTemplateAreas: `
-      "label control control"
-      "label validationIcon validationMessage"
-      "label hint hint"
-      "label . ."
-    `,
-  },
-
-  label: {
-    gridColumnStart: 'label',
-    gridColumnEnd: 'label',
-    gridRowStart: 'label',
-    gridRowEnd: 'label',
-  },
-
-  control: {
-    gridColumnStart: 'control',
-    gridColumnEnd: 'control',
-  },
-
-  validationIcon: {
-    gridColumnStart: 'validationIcon',
-    gridColumnEnd: 'validationIcon',
+    gridTemplateColumns: '33% 1fr',
+    gridTemplateRows: 'auto auto auto 1fr',
   },
 
-  validationMessage: {
-    gridColumnStart: 'validationMessage',
-    gridColumnEnd: 'validationMessage',
-  },
-
-  hint: {
-    gridColumnStart: 'hint',
-    gridColumnEnd: 'hint',
+  // In horizontal layout without a label, replace the label's column with padding.
+  // This lets grid auto-flow properly place the other children, and keeps fields with and without labels aligned.
+  horizontalNoLabel: {
+    paddingLeft: '33%',
+    gridTemplateColumns: '1fr',
```

Comment:
One of the goals of this change is to make it so the children of the `<Field>` component go into the right spot via grid auto flow, without specifying any grid properties on the child. That allows the changes proposed in RFC #26338 to avoid needing a wrapper around the child of `<Field>`.

Reply:
ah, makes sense 👍